enhancements/machine-config/bootc_update_path.md

```diff
@@ -0,0 +1,185 @@
+
+Without On Cluster Layering: 
+osImageURL Update: This change is bootc compatible and bootc enabled. MachineConfigDaemon will read the changes from the new Machine Config. Changes are deployed to the specified nodes by directly shelling out bootc commands. The implementation involves mimicking all the support we have for rpm-ostree and finding bootc equivalent. 
+
+Kernel Argument Update: This change is bootc compatible, but not bootc enabled, meaning that it can only be done via rpm-ostree. Changes picked up by the MachineConfigDaemon will still be sent to the rpm-ostree based update path for landing. The mutated image will be bootc-compatible for further actions.
+
+Extension Change & Kernel Type Switch: This change is bootc incompatible and bootc disabled. These two changes could only be done via rpm-ostree. Worse, these changes will also conflict with bootc, resulting in refusion to future bootc actions. As a result, this limitation can be identified as a regression for switching to bootc update path without On Cluster Layering. However, it can be solved if On Cluster Layering is enabled. The details of which will be discussed below. 
+
+With On Cluster Layering:
+On-Cluster Layering is the next step on MCO’s day-2 configuration journey with the change brought by Image Based RHEL. Upon its GA (planned for 4.17), it will solve the limitation discussed above and become the prerequisite for full bootc compatibility. The MachineOSBuilder will combine the base OS image shipped with OpenShift, any admin-provided custom content, and the MachineConfig content managed by the MCO into a new OCI image. This image will then be deployed to nodes of specified pool using bootc to facilitate a singular, transactional, update. Research has shown that no specific transition is needed as the OCI image built by On Cluster Layering (via Buildah) is by intrinsic bootc compatible. The implementation therefore falls on refactoring the rolling out process. The goal is to ensure that bootc is also able to roll out images built by OCL so that the functionality of OCL remains.
+
+### Workflow Description
+
+1. The cluster administrator wants to specify osImageURL Update / Kernel Argument Update / Extension Change / Kernel Type Switch to the OS on each of their cluster nodes 
+2. The cluster administrator creates new Machine Configs for the changes they want to apply and deploys them to the cluster 
+3. The Machine Config Daemon will read the desired change in the new config
+
+| Type of update | OCL enabled? | Action |
```

Comment:
I think this is trying to describe what happens with and without on cluster layering?

Maybe it would be clearer to split the table headers into:
Type of update
What happens without layering
What happens with layering

Although it might be a bit verbose